filebeat/module/kibana/log/ingest/pipeline-ecs.yml

```diff
@@ -1,29 +1,35 @@
-description: Pipeline for parsing Kibana ecs logs
+description: Pipeline for parsing Kibana ECS logs
 processors:
-- set:
-    field: event.ingested
-    value: '{{_ingest.timestamp}}'
-- set:
-    copy_from: '@timestamp'
-    field: event.created
-- script:
```

Comment:
We need to think how to mirror these changes on the integration packages. We currently use the[ json processor](https://www.elastic.co/guide/en/elasticsearch/reference/master/json-processor.html), which in the end delivers the same result as this script. The problem is that both approaches replace any existing field value with what comes from the logs.

Reply:
another way to write this script to consider existing field values would be:

```java
ctx.json.keySet
  .stream()
  .filter(key -> ctx[key] == null)
  .forEach(key -> ctx[key] = ctx.json.get(key))
```

The problem with this is that Filebeat adds some fields like `log.offset`, `@timestamp`. At this point `ctx[key]` is no longer null. So if a log entry has a `log.logger` field, this script would fail. To make this work with a script would require a more complex code.